plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3select/TestIonSqlQueryBuilder.java

```diff
@@ -106,18 +105,20 @@ public void testDecimalColumns()
                         columns.get(2), Domain.create(ofRanges(Range.range(decimalType, shortDecimal("0.0"), true, shortDecimal("0.02"), true)), false)));
 
         // CSV
-        IonSqlQueryBuilder queryBuilder = new IonSqlQueryBuilder(typeManager, S3SelectDataType.CSV);
+        IonSqlQueryBuilder queryBuilder = new IonSqlQueryBuilder(typeManager, S3SelectDataType.CSV, Optional.empty());
         assertEquals(queryBuilder.buildSql(columns, tupleDomain),
-                "SELECT s._1, s._2, s._3 FROM S3Object s WHERE ((case s._1 when '' then null else CAST(s._1 AS DECIMAL(20,0)) end < 50)) AND " +
-                "(case s._2 when '' then null else CAST(s._2 AS DECIMAL(20,2)) end = 0.05) AND ((case s._3 when '' then null else CAST(s._3 AS DECIMAL(10,2)) " +
-                "end >= 0.00 AND case s._3 when '' then null else CAST(s._3 AS DECIMAL(10,2)) end <= 0.02))");
+                "SELECT s._1, s._2, s._3 FROM S3Object s WHERE " +
```

Comment:
This query is not working as expected for CSV input for S3 Select.
As explained in my previous comment, S3 Select is treating CSV column values as String, hence the `CAST` is needed.

Reply:
I just tried this out with the cast back in and it did fix the query for CSV data but I'm still getting incorrect results for JSON.

Reply:
Would you please provide:
- input JSON file
- query that Trino sends to Select
- expected result?